### PR TITLE
Task/248 improve devops automation

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

_Please delete options that are not relevant._

- [ ] Documentation update
- [ ] Bug fix
- [ ] New metadata support
- [ ] Enhanced metadata
- [ ] Add a CLI option
- [ ] Add something to the core
- [X] Other, please explain: Commit and Doc automation improvements

## What changes did you make? (Give an overview)

- commit messages are automatically templated
- npm run docs is automatically done on all commits

## Is there anything you'd like reviewers to focus on?

double check on your side that this is working correctly

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] ESLint & Prettier are not outputting errors or warnings
- [X] README.md updated (if applicable)
- [X] CHANGELOG.md updated
- [X] ran `npm run docs` to update developer docs

### CODE REVIEW

**Review by JoernBerkefeld — DISMISSED** [review]
lovin it

File: package.json [hunk 1]
```diff
@@ -40,13 +40,13 @@
         "bluebird": "3.7.2",
         "cli-progress": "3.10.0",
         "command-exists": "1.2.9",
-        "conf": "10.1.1",
```
Comment (JoernBerkefeld) [inline package.json:43]: pls dont do that without dedicated PRs per update to track the update

File: package.json [hunk 2]
```diff
@@ -40,13 +40,13 @@
...
+        "conf": "10.1.2",
         "console.table": "0.10.0",
         "fs-extra": "10.0.1",
         "inquirer": "8.2.2",
         "json-to-table": "4.2.1",
         "mustache": "4.2.0",
-        "prettier": "2.6.1",
```
Comment (JoernBerkefeld) [inline package.json:49]: pls dont do that without dedicated PRs per update to track the update

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (DougMidgley) [reply package.json]: reverted

File: package-lock.json [hunk 3]
```diff
@@ -1369,9 +1369,9 @@
             }
         },
         "node_modules/cliui/node_modules/ansi-regex": {
-            "version": "4.1.0",
-            "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-4.1.0.tgz",
-            "integrity": "sha512-1apePfXM1UOSqw0o9IiFAovVz9M5S1Dg+4TrDwfMewQ6p/rmMueb7tWZjQ1rx4Loy1ArBggoqGpfqqdI4rondg==",
```
Comment (JoernBerkefeld) [inline package-lock.json:1374]: can you please revert this file too?

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]
